### PR TITLE
CLN: replace %s syntax with .format in core.indexing and core.internals

### PR DESCRIPTION
**Pull request by AaronCritchley**
- [x] Progress towards #16130
- [x] tests added / passed
- [x] passes git diff upstream/master -u -- "*.py" | flake8 --diff

### CODE REVIEW

File: pandas/core/internals.py [hunk 1]
```diff
@@ -634,11 +637,13 @@ def _astype(self, dtype, copy=False, errors='raise', values=None,
...
         if newb.is_numeric and self.is_numeric:
             if newb.shape != self.shape:
-                raise TypeError("cannot set astype for copy = [%s] for dtype "
-                                "(%s [%s]) with smaller itemsize that current "
-                                "(%s [%s])" % (copy, self.dtype.name,
-                                               self.itemsize, newb.dtype.name,
-                                               newb.itemsize))
+                raise TypeError(
+                    "cannot set astype for copy = [{copy}] for dtype "
+                    "({dtype} [{itemsize}]) with smaller itemsize that "
```
Comment (AaronCritchley) [inline pandas/core/internals.py]: Is this supposed to be 'smaller itemsize **than** current'? If yes, would you like me to change in this PR?

Comment (jreback) [reply pandas/core/internals.py]: yes

**Comment by jreback** [comment]
lgtm. ping when green (after than tiny change).

**Comment by AaronCritchley** [comment]
@jreback last build green, change made to the error message as discussed 😄

**Comment by jreback** [comment]
thanks!

**Pull request merged**